### PR TITLE
[trainer] add option to ignore keys for the train function too (#11719)

### LINKED ISSUE #11719: parameter `ignore_keys` of `trainer.predict` not accessible in `Trainer` or `TrainingArguments`
# 🚀 Feature request

The [`predict`](https://huggingface.co/transformers/main_classes/trainer.html#transformers.Trainer.predict) and [`evaluate`](https://huggingface.co/transformers/main_classes/trainer.html#transformers.Trainer.evaluate) methods of the Trainer class provide an excellent option of `ignore_keys`. Here is a small example:

```python
trainer.predict(dataset, ignore_keys=["ner_loss", "cls_loss", "ner_logits", "cls_logits"])
```

@sgugger 

This option is however, not accessible during the normal setup of defining `TrainingArguments` class nor the `Trainer` class so the a call to `trainer.train()` leads to errors during the mid-training evaluation.

## Motivation

I am unable to evaluate the model metrics on the validation set **during** the training to see if makes sense to continue.

## Your contribution

I am happy to make a PR if this is seen as a genuine problem. Like always, may be I am missing something.

### PR DESCRIPTION
**Pull request by shabie**
# What does this PR do?

This pull request adds the option of ignoring certain output keys for evaluation during the training phase. As of now, this option is only available for `predict` and `evaluation` methods of the `Trainer` class which can only be called after the training.

Fixes #11719

Changes:

1. Add a new parameter to the `Trainer.train` function called `ignore_keys_for_eval`.
2. Pass this to the `ignore_keys` parameter of the `trainer.evaluate` function that is already called within the `trainer.train` function.
2. Add the parameter to the docstring.

### CODE REVIEW

**Review by sgugger — APPROVED** [review]
Thanks for adding this!
